storage/src/vespa/storage/distributor/operations/external/putoperation.cpp

```diff
@@ -129,29 +134,59 @@ bool PutOperation::at_least_one_storage_node_is_available() const {
     return false;
 }
 
+bool PutOperation::has_condition() const noexcept {
+    return _msg->hasTestAndSetCondition();
+}
+
 void
 PutOperation::onStart(DistributorStripeMessageSender& sender)
 {
-    document::BucketIdFactory bucketIdFactory;
-    document::BucketId bid = bucketIdFactory.getBucketId(_msg->getDocumentId());
+    LOG(debug, "Received Put %s for bucket %s",
+        _msg->getDocumentId().toString().c_str(), _doc_id_bucket_id.toString().c_str());
+
+    if (!has_condition()) {
+        start_direct_put_dispatch(sender);
+    } else {
+        start_conditional_put(sender);
+    }
+}
+
+void PutOperation::start_conditional_put(DistributorStripeMessageSender& sender) {
+    document::Bucket bucket(_msg->getBucket().getBucketSpace(), _doc_id_bucket_id);
+    _check_condition = CheckCondition::create_if_inconsistent_replicas(bucket, _bucket_space, _msg->getDocumentId(),
+                                                                       _msg->getCondition(), _node_ctx, _op_ctx,
+                                                                       _temp_metric);
+    if (!_check_condition) {
+        start_direct_put_dispatch(sender);
+    } else {
+        // Inconsistent or non-existent replicas; might need write repair
```

Comment:
// Inconsistent replicas; need write repair

Reply:
Updated comment (and another comment below)